### PR TITLE
+ option to disable the connection pooling

### PR DESCRIPTION
**Pull request by akurilov**

### CODE REVIEW

File: src/main/java/com/emc/mongoose/storage/driver/pravega/PravegaStorageDriver.java [hunk 1]
```diff
@@ -363,6 +360,12 @@ public PravegaStorageDriver(
 			.findFirst()
 			.orElseGet(this::getBasicCredentialsIfAny);
 		Loggers.MSG.info("Authentication type: {}", cred == null ? "none" : cred.getAuthenticationType());
+		val connConfig = nodeConfig.configVal("conn");
+		val poolingFlag = connConfig.boolVal("pooling");
+		evtWriterConfig = EventWriterConfig
+				.builder()
+				.enableConnectionPooling(poolingFlag)
+				.build();
```
Comment (dadlex) [inline src/main/java/com/emc/mongoose/storage/driver/pravega/PravegaStorageDriver.java:368]: is it intentional not to include backOffMillis and retryAttempts?

**Review by dadlex — APPROVED** [review]

File: src/main/java/com/emc/mongoose/storage/driver/pravega/PravegaStorageDriver.java [hunk 2]
```diff
@@ -707,9 +710,11 @@ void createEventsTransaction(final List<O> ops) {
 				// create the batch event stream writer if necessary
 				val txnEvtWriterCache = threadLocalTxnEvtWriterCache.get();
 				val batchEvtWriter = txnEvtWriterCache.computeIfAbsent(
-								clientFactory,
-								clientFactory_ -> clientFactory_.createTransactionalEventWriter(
-												streamName, evtSerializer, evtWriterConfig));
+					clientFactory,
+					clientFactory_ -> clientFactory_.createTransactionalEventWriter(
+						UUID.randomUUID().toString(), streamName, evtSerializer, evtWriterConfig
```
Comment (dadlex) [inline src/main/java/com/emc/mongoose/storage/driver/pravega/PravegaStorageDriver.java:723]: I've seen both Long.toString(System.nanoTime()) and now UUID.randomUUID().toString() to create a unique ID for an instance. Is there a general guideline when to use which?

**Review by dadlex — APPROVED** [review]

File: src/main/java/com/emc/mongoose/storage/driver/pravega/PravegaStorageDriver.java (hunk 1 above)
Comment (akurilov) [reply src/main/java/com/emc/mongoose/storage/driver/pravega/PravegaStorageDriver.java:368]: Yep, they're missing in mongoose config, so the defaults should be used

File: src/main/java/com/emc/mongoose/storage/driver/pravega/PravegaStorageDriver.java (hunk 2 above)
Comment (akurilov) [reply src/main/java/com/emc/mongoose/storage/driver/pravega/PravegaStorageDriver.java:723]: This call uses `UUID.randomUUID().toString()` internally if the arg is omitted, so I just left this the same

File: src/main/java/com/emc/mongoose/storage/driver/pravega/PravegaStorageDriver.java (hunk 1 above)
Comment (dadlex) [reply src/main/java/com/emc/mongoose/storage/driver/pravega/PravegaStorageDriver.java:368]: Do we want them in the config? Are those expected to be varied?

**Review by AndreyKoltsov1997 — APPROVED** [review]

File: src/main/java/com/emc/mongoose/storage/driver/pravega/PravegaStorageDriver.java (hunk 1 above)
Comment (akurilov) [reply src/main/java/com/emc/mongoose/storage/driver/pravega/PravegaStorageDriver.java:368]: Not sure that they're needed now. We will add them if needed.

**Pull request merged**